Initial version (#1)
SumoLogic/sumologic-opentelemetry-python: README.md

- [opentelemetry-instrumentation-pymysql](https://pypi.org/project/opentelemetry-instrumentation-pymysql)
- [opentelemetry-instrumentation-pyramid](https://pypi.org/project/opentelemetry-instrumentation-pyramid)
- [opentelemetry-instrumentation-redis](https://pypi.org/project/opentelemetry-instrumentation-redis)
- [opentelemetry-instrumentation-requests](https://pypi.org/project/opentelemetry-instrumentation-requests)
- [opentelemetry-instrumentation-sklearn](https://pypi.org/project/opentelemetry-instrumentation-sklearn)
- [opentelemetry-instrumentation-sqlalchemy](https://pypi.org/project/opentelemetry-instrumentation-sqlalchemy)
- [opentelemetry-instrumentation-starlette](https://pypi.org/project/opentelemetry-instrumentation-starlette)
- [opentelemetry-instrumentation-tornado](https://pypi.org/project/opentelemetry-instrumentation-tornado)
- [opentelemetry-instrumentation-urllib3](https://pypi.org/project/opentelemetry-instrumentation-urllib3)

### Included propagators

```
opentelemetry-distro
opentelemetry-instrumentation
opentelemetry-exporter-otlp-proto-http
opentelemetry-propagator-jaeger
opentelemetry-propagator-b3
opentelemetry-propagator-aws-xray

> opentelemetry-propagator-ot-trace

Reviewer: please make it a list with links like `### Instrumented packages`

```

## Application execution

>  Execute the following command. The command will also print the opentelemetry config used.

Reviewer: ```suggestion
Execute the following command. The command will also print the opentelemetry config used.
```

```
sumologic-opentelemetry-instrument python3 SCRIPT_NAME.py
```

## Environment variables config

The wrapper command checks for the following basic OpenTelemetry env variables that are required for the collected data to be useful within Sumo Logic:

- `OTEL_PROPAGATORS`
- `OTEL_TRACES_EXPORTER`
- `OTEL_SERVICE_NAME`
- `OTEL_EXPORTER_OTLP_ENDPOINT`
- `OTEL_RESOURCE_ATTRIBUTES`

### Propagators

By default, all propgators are enabled (`OTEL_PROPAGATORS=tracecontext,baggage,b3,b3multi,jaeger,xray,ottrace`).

### Exporter.

By default, exported is set to OTLP HTTP (`OTEL_TRACES_EXPORTER=otlp_proto_http`)

### Service name

> By default, service name will not be set `OTEL_SERVICE_NAME=`. Overwrite the env var with a string value representing service business logic, such as "FinanceServiceCall". This will appear as a tracing service name in Sumo Logic.

Reviewer: ```suggestion
By default, service name will not be set `OTEL_SERVICE_NAME=`.
Overwrite the env var with a string value representing service business logic, such as "FinanceServiceCall".
This will appear as a tracing service name in Sumo Logic.
```

> 
> 

Reviewer: ```suggestion

```

### Endpoint

> By default, endpoint is not set (`OTEL_EXPORTER_OTLP_ENDPOINT=`). Represents the endpoint where telemetry data will be sent. Change to appropriate endpoint, for example `OTEL_EXPORTER_OTLP_ENDPOINT=http://collection-sumologic-otelcol.sumologic:55681`

Reviewer: ```suggestion
By default, endpoint is not set (`OTEL_EXPORTER_OTLP_ENDPOINT=`). Represents the endpoint where telemetry data will be sent.
Change to appropriate endpoint, for example `OTEL_EXPORTER_OTLP_ENDPOINT=http://collection-sumologic-otelcol.sumologic:55681`
```

# Resource attributes

> By default, resource attributes are not set (`OTEL_RESOURCE_ATTRIBUTES=`). Is used to configure the application name (i.e. `OTEL_RESOURCE_ATTRIBUTES=application=APPLICATION_NAME`). The application name will appear as a tracing application name in Sumo Logic. Additional attributes can be added here as comma separated key=value pairs.

Reviewer: ```suggestion
By default, resource attributes are not set (`OTEL_RESOURCE_ATTRIBUTES=`).
Is used to configure the application name (i.e. `OTEL_RESOURCE_ATTRIBUTES=application=APPLICATION_NAME`).
The application name will appear as a tracing application name in Sumo Logic.
Additional attributes can be added here as comma separated key=value pairs.
```

# Sumo Logic OpenTelemetry Python

> 
> 

Reviewer: ```suggestion

```

An all-in-one package for python projects used to enable OpenTelemetry auto-instrumentation.

It contains all supported propagators and auto-instrumentation plugins.

> ## Instalation

Reviewer 2: ```suggestion
## Installation
```

```
pip install sumologic-opentelemetry
```

> 
> 

Reviewer: ```suggestion

```

## Instrumented packages

> This package installs all officially supported auto-instrumentation packages() as well as comonly used propagators and exporters:

Reviewer: ```suggestion
This package installs all officially supported auto-instrumentation packages as well as commonly used propagators and exporters:
```

opentelemetry-instrumentation-wsgi
opentelemetry-instrumentation-aiohttp-client
opentelemetry-instrumentation-aiopg
opentelemetry-instrumentation-asgi
opentelemetry-instrumentation-asyncpg
opentelemetry-instrumentation-boto
opentelemetry-instrumentation-botocore
opentelemetry-instrumentation-celery
opentelemetry-instrumentation-django
opentelemetry-instrumentation-elasticsearch
opentelemetry-instrumentation-falcon
opentelemetry-instrumentation-fastapi
opentelemetry-instrumentation-flask
opentelemetry-instrumentation-grpc
opentelemetry-instrumentation-httpx
opentelemetry-instrumentation-jinja2
opentelemetry-instrumentation-kafka-python
opentelemetry-instrumentation-mysql
opentelemetry-instrumentation-pika
opentelemetry-instrumentation-psycopg2
opentelemetry-instrumentation-pymemcache
opentelemetry-instrumentation-pymongo
opentelemetry-instrumentation-pymysql
opentelemetry-instrumentation-pyramid
opentelemetry-instrumentation-redis
opentelemetry-instrumentation-requests
opentelemetry-instrumentation-sklearn
opentelemetry-instrumentation-sqlalchemy
opentelemetry-instrumentation-starlette
opentelemetry-instrumentation-tornado
opentelemetry-instrumentation-urllib3

> ```

Reviewer: I would make it list with links to code/documentation

Author: Is redirecting to pypi packages enough? 
Most packages have pointer to the documentation, i.e. https://pypi.org/project/opentelemetry-instrumentation-urllib3/

Reviewer: Yes, pointing to these docs should be fine

Reviewer: Sorry, I thought You were meaning to link every module separately e.g. `[opentelemetry-instrumentation-urllib3](https://pypi.org/project/opentelemetry-instrumentation-urllib3/)`

Author: It is done separatly.

### Included propagators
```
opentelemetry-distro
opentelemetry-instrumentation
opentelemetry-exporter-otlp-proto-http
opentelemetry-propagator-jaeger
opentelemetry-propagator-b3
opentelemetry-propagator-aws-xray
opentelemetry-propagator-ot-trace
```

## Application execution

>  Execute the following command. The command will also print the opnteleemtry config used.

Reviewer 2: ```suggestion
 Execute the following command. The command will also print the opentelemetry config used.
```

```
sumologic-opentelemetry-instrument python3 SCRIPT_NAME.py
```

## Environment variables config

> The wrapper command checks for the basic opentelemetry env variables - OTEL_PROPAGATORS, OTEL_TRACES_EXPORTER, OTEL_SERVICE_NAME, OTEL_EXPORTER_OTLP_ENDPOINT, OTEL_RESOURCE_ATTRIBUTES - that are required for the collected data to be usefull withinn Sumo Logic.

Reviewer: ```suggestion
The wrapper command checks for the following basic OpenTelemetry env variables that are required for the collected data to be useful within Sumo Logic:

- `OTEL_PROPAGATORS`
- `OTEL_TRACES_EXPORTER`
- `OTEL_SERVICE_NAME`
- `OTEL_EXPORTER_OTLP_ENDPOINT`
- `OTEL_RESOURCE_ATTRIBUTES`
```

If those are set, the wrapper will not attempt to overwrite them.

### Propagators

> By default all propgators are enables (`OTEL_PROPAGATORS=tracecontext,baggage,b3,b3multi,jaeger,xray,ottrace`)

Reviewer: ```suggestion
By default all propgators are enabled (`OTEL_PROPAGATORS=tracecontext,baggage,b3,b3multi,jaeger,xray,ottrace`)
```

> ### Exporter.

Reviewer: ```suggestion
### Exporter
```

> ### Exporter.
> By default exported is set to OTLP HTTP (`OTEL_TRACES_EXPORTER=otlp_proto_http`)

Reviewer: ```suggestion
### Exporter

By default exporter is set to OTLP HTTP (`OTEL_TRACES_EXPORTER=otlp_proto_http`).
```

### Service name

> By default service name will not be set `OTEL_SERVICE_NAME=`. Overwrite the env var with a string value representing service business logic, such as "FinanceServiceCall". This will appear as a tracing service name in Sumo Logic.

Reviewer 2: ```suggestion
By default, service name will not be set `OTEL_SERVICE_NAME=`. Overwrite the env var with a string value representing service business logic, such as "FinanceServiceCall". This will appear as a tracing service name in Sumo Logic.
```

### Endpoint

> By default endpoint is not set (`OTEL_EXPORTER_OTLP_ENDPOINT=`). Represents the endpoint where telemetry data will be sent. Change to appropriate endpoint, for example `OTEL_EXPORTER_OTLP_ENDPOINT=http://collection-sumologic-otelcol.sumologic:55681`

Reviewer 2: ```suggestion
By default, endpoint is not set (`OTEL_EXPORTER_OTLP_ENDPOINT=`). Represents the endpoint where telemetry data will be sent. Change to appropriate endpoint, for example `OTEL_EXPORTER_OTLP_ENDPOINT=http://collection-sumologic-otelcol.sumologic:55681`
```

# Resource attributes

> By default not set (`OTEL_RESOURCE_ATTRIBUTES=`). Is used to configure the application name (i.e. `OTEL_RESOURCE_ATTRIBUTES=application=APPLICATION_NAME`). The application name will appear as a tracing application name in Sumo Logic. Additional attributes can be added here as comma separated key=value pairs.

Reviewer 2: ```suggestion
By default, resource attributes are not set (`OTEL_RESOURCE_ATTRIBUTES=`). Is used to configure the application name (i.e. `OTEL_RESOURCE_ATTRIBUTES=application=APPLICATION_NAME`). The application name will appear as a tracing application name in Sumo Logic. Additional attributes can be added here as comma separated key=value pairs.
```
